frontend/packages/console-shared/src/utils/promise-helper.ts

```diff
@@ -0,0 +1,37 @@
+const promiseResolver = <T extends unknown>(
+  promise: Promise<T>,
+  onSuccess: VoidFunction,
+  onFailure: VoidFunction,
+) => {
+  promise.then((x) => onSuccess(x)).catch((e) => onFailure(e));
+};
+
+export const retryPromise = <T extends unknown>(
```

Comment:
Is there a reason we can't use a watch with a `metadata.name` filter instead? Then you don't need retries, and the data appears immediately when it's created.

Reply:
We could but for a simple object like secret and ConfigMap is it worth polling over and over? It would get created after a second or two.

Reply:
I think it's a lot cleaner and more robust. It won't fail if the resource isn't created in whatever arbitrary number of retries we pick.

We use watches all over console. If we're worried about overhead, we should be looking at places like the Home -> Status page or Topology, which hold a lot of expensive watches. Watching a single secret or config map is negligible.

I'm concerned with adding a generic retry helper to shared because it's not a pattern we should encourage.

Reply:
ack removing this helper and using watch.